metadata/src/main/java/com/redhat/lightblue/rest/metadata/hystrix/ReIndexCommand.java

```diff
@@ -0,0 +1,47 @@
+/**
+ *
+ */
+package com.redhat.lightblue.rest.metadata.hystrix;
+
+import com.redhat.lightblue.crud.CRUDController;
+import com.redhat.lightblue.metadata.EntityMetadata;
+import com.redhat.lightblue.metadata.Metadata;
+import com.redhat.lightblue.mongo.crud.MongoCRUDController;
+import com.redhat.lightblue.rest.RestConfiguration;
+
+/**
+ * @author bvulaj
+ *
+ */
+public class ReIndexCommand extends AbstractRestCommand {
+
+    private final String entity;
+    private final String version;
+
+    public ReIndexCommand(String clientKey, Metadata metadata, String entity, String version) {
+        super(ReIndexCommand.class, clientKey, metadata);
+        this.entity = entity;
+        if ("default".equals(version)) {
+            this.version = null;
+        } else {
+            this.version = version;
+        }
+    }
+
+
+    /* (non-Javadoc)
+     * @see com.netflix.hystrix.HystrixCommand#run()
+     */
+    @Override
+    protected String run() throws Exception {
+        EntityMetadata emd = getMetadata().getEntityMetadata(entity, version);
```

Comment:
When version is removed, you can use Factory.getCRUDController(ei.getDataStore().getBackend())